src/views/open-api-docs-view-v2/types.ts

```diff
@@ -3,28 +3,41 @@
  * SPDX-License-Identifier: MPL-2.0
  */
 
-import { OperationContentProps } from './components/operation-content'
-import { LandingContentProps } from './components/landing-content'
-
-/**
- * Nav items are used to render the sidebar.
- *
- * TODO: will likely need to be expanded once a more fully-featured sidebar
- * is constructed. Older OpenAPI docs view may be useful for prior art, though
- * we probably don't need to feel too tied to it.
- */
-export type OpenApiNavItem = {
-	title: string
-	fullPath: string
-	isActive: boolean
-}
+// Types
+import type { LandingContentProps } from './components/landing-content'
+import type { OpenAPIV3 } from 'openapi-types'
+import type { OperationContentProps } from './components/operation-content'
+import type { OperationObject } from './utils/get-operation-objects'
+import type { ProductSlug } from 'types/products'
 
 /**
  * Shared props are common to both the "landing" and "operation" views.
  */
 export interface SharedProps {
```

Comment:
In general I've opted to avoid reaching for type abstractions, as in the past I think we've been a bit overzealous in trying to create shared types, resulting in highly entangled components that can be difficult to iterate on. I find it a lot easier to declare things explicitly.

The downside is it can feel a bit repetitive, but I find it makes it much easier to change things without worrying about side-effects elsewhere in the app. And of course, TypeScript will still do the job of warning us about places that'll be affected by any type changes, regardless of how fancy our type abstractions are.